miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultProjectService.java

```diff
@@ -194,12 +203,14 @@ private void loadChildEntities(Project project) throws IOException {
         targetedSequencingService,
         "defaultTargetedSequencingId");
     loadChildEntity(project::setPipeline, project.getPipeline(), pipelineService, "pipelineId");
-    Set<ProjectContactsAndRole> contacts = new HashSet<>();
-    for (ProjectContactsAndRole element : project.getContacts()) {
+    Set<ProjectContact> contacts = new HashSet<>();
+    for (ProjectContact element : project.getContacts()) {
       Contact contact = contactService.get(element.getContact().getId());
       ContactRole contactRole = contactRoleService.get(element.getContactRole().getId());
-      contacts
-          .add(new ProjectContactsAndRole(project, contact, contactRole));
+      // check to ensure the values load properly, and if they do, add it to the set
+      if (contact != null && contactRole != null) {
+        contacts.add(element);
+      }
```

Comment:
* This will ignore invalid contact or role IDs. Use `loadChildEntity`, which will handle this properly
* You're removing all of the `ProjectContacts`s from the project, and then re-adding the exact same objects. There's no need to create a new collection or add/remove items from the existing one

Reply:
Since there isn't a setter method for contact and contact roles for project (to put as a parameter for `loadChildEntities`), I decided to just follow a similar method to how assay tests and metrics were handled for validation.

Reply:
The setter doesn't have to be on the project:

```
for (ProjectContact projectContact : project.getContacts()) {
  loadChildEntity(projectContact::setContact, projectContact.getContact(), contactService, "contact");
  loadChildEntity(projectContact::setContactRole, projectContact.getContactRole(), contactRoleService, "contact");
}
```

Note "contact" is used as property name for both so that the error messages land in the same `projectForm_contactErrors` div